src/v2/Utils/Hooks/useEventTiming.tsx

```diff
@@ -0,0 +1,88 @@
+import { DateTime, Duration } from "luxon"
+import { useMemo } from "react"
+
+const NBSP = " "
+const SEPARATOR = `${NBSP}:${NBSP}`
+
+const padWithZero = (n: number) => {
+  return n.toString().padStart(2, "0")
+}
+
+interface UseEventTiming {
+  startAt: string
+  endAt: string
+  currentTime: string
+}
+
+export const useEventTiming = ({
+  currentTime,
+  startAt,
+  endAt,
+}: UseEventTiming) => {
+  const durationTilEnd = Duration.fromISO(
+    DateTime.fromISO(endAt).diff(DateTime.fromISO(currentTime)).toString()
+  )
+  const daysTilEnd = durationTilEnd.as("days")
+  const secondsTilEnd = durationTilEnd.as("seconds")
+
+  const hasStarted =
+    Duration.fromISO(
+      DateTime.fromISO(startAt).diff(DateTime.fromISO(currentTime)).toString()
+    ).seconds < 0
+
+  const hasEnded = Math.floor(secondsTilEnd) <= 0
+  const closesSoon = daysTilEnd <= 3 && daysTilEnd >= 1
```

Comment:
Minor bug #1: Used to be `daysTilEnd > 1` vs `>=` meaning if it was exactly a day, the formatted time would be null.